File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const pObj = {}
    for(const p of participant){
        pObj[p] = pObj[p]!==undefined ? pObj[p]+1 : 0
    }
    
    for(const c of completion){
        pObj[c] -= 1
    }
    
    for(const [k,v] of Object.entries(pObj)){
        if(v===0){
            return k
        }
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        Map<String, Integer> map = new HashMap<>();
        
        for(String p : participant) {
            if(map.containsKey(p)){
                map.put(p, map.get(p)+1);
                continue;
            }
            map.put(p, 1);
        }
        
        for(String c : completion) {
            map.put(c, map.get(c)-1);
        }
        
        for(String k: map.keySet()) {
            if(map.get(k)==1) return k;
        }
        
        return "";
    }
}